Drivers/imu/inc/icm42688.hpp

```diff
@@ -0,0 +1,122 @@
+// icm42688.hpp
+
+#ifndef ICM42688_HPP
+#define ICM42688_HPP
+
+#include "imu_driver.hpp"
+#include <stdint.h>
+#include "stm32l5xx_hal.h"
+#include "stm32l5xx_hal_gpio.h"
+#include <stm32l552xx.h>
+
+#define GYRO_CALIBRATION_DATA_BUFFER_SIZE 14
+#define GYRO_CALIBRATION_RAW_DATA_BUFFER_SIZE 7
+#define RAW_DATA_BUFFER_SIZE 7
+
+class ICM42688 : public IMUDriver {
+    public:
+        ICM42688(SPI_HandleTypeDef * SPI_HANDLE, GPIO_TypeDef * CS_GPIO_PORT, uint16_t CS_PIN);
+
+        /**
+        * Sets up the IMU's initial starting conditions by setting 
+        * the cs pin and calling reset(), setLowNoiseMode(), and calibrate()
+         *
+        * @return none
+         */
+        void beginMeasuring() override;
+        
+        /**
+        * Calibrates the gyroscope. Algorithm adapted from https://github.com/finani/ICM42688/tree/master/src
+         *
+        *  @return none 
+         */
+        void calibrate() override;
+        
+        /**
+        * Retrieve and process IMU accelerometer and gyroscope data into m/s and deg/s
+         *
+        * @param data_buffer -> array used to store and process raw data from IMU
+         * 
+        * @return IMUData_t structure consisting of accelerometer and gyroscope data in m/s and deg/s
+         */
+        IMUData_t getResult(uint8_t * data_buffer) override;
+
+    private:
+        /**
+        * Communicate with IMU via SPI to read raw data
+         * 
+        * @param sub_address -> memory address of starting byte to be retrieved
+        * @param count -> number of bytes to be retrieved
+        * @param dest -> array to be populated with raw data
+         * 
+        * @return none
+         */
+        void readRegister(uint8_t sub_address, uint8_t count, uint8_t * dest);
```

Comment:
nit: `count` is not as descriptive as `num_bytes_to_retrieve`. It's not toooo much longer.

Also, sub_address and dest, try and use concatenation either or don't use it at all.